### PR TITLE
Fix #37 ident signature in Union component of ui-legacy-router

### LINKED ISSUE #37: legacy-ui-routers generate invalid component
```
(do
  (comp/defsc Users [this {:PAGE/keys [ident id]}]
    {:query         [:PAGE/ident :PAGE/id]
     :ident         (fn [] [ident id])
     :initial-state (fn [_]
                      {:PAGE/ident :PAGE/users :PAGE/id :PAGE/users})})
  (fr/defsc-router RootRouter [this {:PAGE/keys [ident id]}]
    {:default-route  Users
     :ident          (fn [] [ident id])

     :router-targets {:PAGE/users Users}
     :router-id      :PAGE/root-router})
  (comp/defsc Root [this _]
    {:query         [{:root-router (comp/get-query RootRouter)}]
     :initial-state (fn [& _]
                      {:root-router (comp/get-initial-state RootRouter _)})})
  (fnorm/tree->db Root
                  (comp/get-initial-state Root {})
                  true))
```
^This code throws
```
Execution error (ArityException) at com.fulcrologic.fulcro.components/ident (components.cljc:161).
Wrong number of args (2) passed to: user/eval19796/ident*--19798
```

### PR DESCRIPTION
**Pull request by souenzzo**

### CODE REVIEW

**Comment by abhi18av** [comment]
Hii @souenzzo  , is this issue related to the issue created here  https://github.com/fulcrologic/fulcro3/issues/37 ?

**Comment by souenzzo** [comment]
Yes @abhi18av

File: src/test/com/fulcrologic/fulcro/routing/legacy_ui_routers_spec.cljc [hunk 1]
```diff
@@ -18,10 +19,24 @@
    :router-targets {:PAGE/simple-target SimpleTarget}
    :router-id      :PAGE/root-router})
 
+(comp/defsc Root [this _]
+          {:query         [{:root-router (comp/get-query SimpleTarget)}]
+           :initial-state (fn [& _]
```
Comment (awkay) [inline src/test/com/fulcrologic/fulcro/routing/legacy_ui_routers_spec.cljc]: I would prefer not to advertise this as a "correct" way to spec args for initial state.

File: src/test/com/fulcrologic/fulcro/routing/legacy_ui_routers_spec.cljc [hunk 2]
```diff
@@ -18,10 +19,24 @@
...
+
 (specification "defsc-router Macro"
   (component "Basic feature access"
     (assertions
       "Just returns it's query"
       (comp/get-query SimpleRouter)
       => [::fr/id
-          {::fr/current-route {:PAGE/simple-target [:PAGE/id :PAGE/ident]}}])))
+          {::fr/current-route {:PAGE/simple-target [:PAGE/id :PAGE/ident]}}]))
+  (component "Normalization"
```
Comment (awkay) [inline src/test/com/fulcrologic/fulcro/routing/legacy_ui_routers_spec.cljc]: So, I would prefer tests that only involve the items under test.  In the case you're reporting, a compile error will result if the macro is incorrect.  Therefore, simple defining a router should reproduce the error.  If you want to add a test of something, then a test like: "router initial state can be defined and read" where you've defined it, allow it to take params, USE the params in initial state, and verify that the map that comes back uses the param.

Drop the whole `tree->db` stuff...it is noise and makes the test harder to understand.

File: src/test/com/fulcrologic/fulcro/routing/legacy_ui_routers_spec.cljc [hunk 3]
```diff
@@ -18,10 +19,24 @@
    :router-targets {:PAGE/simple-target SimpleTarget}
    :router-id      :PAGE/root-router})
 
+(comp/defsc Root [this _]
+          {:query         [{:root-router (comp/get-query SimpleTarget)}]
```
Comment (awkay) [inline src/test/com/fulcrologic/fulcro/routing/legacy_ui_routers_spec.cljc]: I don't think you mean SimpleTarget...I think you want the router.

File: src/test/com/fulcrologic/fulcro/routing/legacy_ui_routers_spec.cljc (hunk 1 above)
Comment (souenzzo) [reply src/test/com/fulcrologic/fulcro/routing/legacy_ui_routers_spec.cljc]: Not Desired. Will remove

**Comment by awkay** [comment]
Thanks!

**Pull request merged**